pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -381,7 +381,9 @@ def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
     return x
 
 
-def _update_n(bar: _tqdm, value: int) -> None:
-    if not bar.disable:
-        bar.n = value
-        bar.refresh()
+def _update_n(bar: _tqdm, current: int, refresh_rate: int) -> None:
+    total = bar.total
+    leftover = current % refresh_rate
+    advance = leftover if (current == total and leftover != 0) else refresh_rate
```

Comment:
leftover is only used when `current == total`

and current is only used for `leftover`

which means current could be replaced by `total` and we wouldn't need to pass `current`

Reply:
how would you find the condition for `current == total` without `current`? at the end of the progress bar, we need to ensure that it updates for the `leftover` steps if the refresh rate is higher.